server/src/test/java/keywhiz/service/resources/BatchSecretDeliveryResourceIntegrationTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package keywhiz.service.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import io.dropwizard.jackson.Jackson;
+import keywhiz.IntegrationTestRule;
+import keywhiz.KeywhizService;
+import keywhiz.TestClients;
+import keywhiz.api.ApiDate;
+import keywhiz.api.BatchSecretRequest;
+import keywhiz.api.model.Secret;
+import keywhiz.client.KeywhizClient;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+import static keywhiz.testing.HttpClients.testUrl;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class BatchSecretDeliveryResourceIntegrationTest {
```

Comment:
Good tests! A couple of notes:
- Please document why "General_Password" is accessible to this client, but "Hacking_Password" isn't (or just create new secrets for this test).
- Please add a test of retrieving multiple secrets which succeed, and multiple secrets where one succeeds and another fails (to make sure that the whole request fails in that case)

Reply:
I hadn't yet read the mocked test, which already tests multiple secrets which succeed :) Alternatively, to keep the integration test lightweight, we could add the test with multiple secrets where one succeeds and the other fails to the mocked test. With a new test for AclDAO to make sure that retrieving batched secrets works as expected, that should be sufficient. 

(Documentation of why these secrets are accessible/inaccessible would still be good, though.)